src/Functional/Group.php

```diff
@@ -11,14 +11,19 @@
 namespace Functional;
 
 use Functional\Exceptions\InvalidArgumentException;
-use Traversable;
 
 /**
  * Groups a collection by index returned by callback.
  *
- * @param Traversable|array $collection
- * @param callable $callback
- * @return array
+ * @template K of array-key
+ * @template V
+ * @template G of array-key
+ *
+ * @param iterable<K,V> $collection
+ * @param callable(V,K,iterable<K,V>):G $callback
+ *
+ * @return (G is numeric-string ? array<int,array<K,V>> : array<G,array<K,V>>)
```

Comment:
I don't think this is possible right now, see https://github.com/phpstan/phpstan/issues/6847

Also the numeric-string isn’t correct, because this:

```php
["1.2" => "foo"];
```

The key is a numeric string but the result will be a string array, not an integer array.

Reply:
You got a very good point and this is very sad since I'll have to remove this in a lot of functions :cry: